src/net_processing.cpp

```diff
@@ -5252,15 +5287,22 @@ void PeerManagerImpl::CheckForStaleTipAndEvictPeers()
     EvictExtraOutboundPeers(now);
 
     if (now > m_stale_tip_check_time) {
-        // Check whether our tip is stale, and if so, allow using an extra
-        // outbound peer
-        if (!m_chainman.m_blockman.LoadingBlocks() && m_connman.GetNetworkActive() && m_connman.GetUseAddrmanOutgoing() && TipMayBeStale()) {
-            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n",
-                      count_seconds(now - m_last_tip_update.load()));
-            m_connman.SetTryNewOutboundPeer(true);
-        } else if (m_connman.GetTryNewOutboundPeer()) {
-            m_connman.SetTryNewOutboundPeer(false);
+        // Skip checks when blocks are being loaded from disk or the network is disabled
+        if (!m_chainman.m_blockman.LoadingBlocks() && m_connman.GetNetworkActive()) {
+            // Check whether our tip is stale, and if so, allow using an extra outbound peer
+            if (m_connman.GetUseAddrmanOutgoing() && TipMayBeStale()) {
+                LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n",
+                          count_seconds(now - m_last_tip_update.load()));
+                m_connman.SetTryNewOutboundPeer(true);
+            } else if (m_connman.GetTryNewOutboundPeer()) {
+                m_connman.SetTryNewOutboundPeer(false);
```

Comment:
The conditions under which `m_connman.SetTryNewOutboundPeer(false);` is executed are not the same before and after this change. Maybe it is ok, but is an unrelated change to the aim of this PR:

* before `m_connman.SetTryNewOutboundPeer(false)` would have been executed if this was true:
```cpp
(m_chainman.m_blockman.LoadingBlocks() || !m_connman.GetNetworkActive() || !m_connman.GetUseAddrmanOutgoing() || !TipMayBeStale()) && m_connman.GetTryNewOutboundPeer()
```

* after `m_connman.SetTryNewOutboundPeer(false)` will be executed if this is true:
```cpp
!m_chainman.m_blockman.LoadingBlocks() && m_connman.GetNetworkActive() && (!m_connman.GetUseAddrmanOutgoing() || !TipMayBeStale()) && m_connman.GetTryNewOutboundPeer()
```

Hopefully this entire hunk can be [dropped](https://github.com/bitcoin/bitcoin/pull/28170/files#r1434937507).

Reply:
Dropped per comment https://github.com/bitcoin/bitcoin/pull/28170#discussion_r1434937507. Thanks